testing/src/test/java/io/stargate/it/http/RestApiv2DseTest.java

```diff
@@ -0,0 +1,154 @@
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.CqlSessionSpec;
+import io.stargate.it.driver.TestKeyspace;
+import io.stargate.it.http.models.Credentials;
+import io.stargate.it.storage.StargateConnectionInfo;
+import io.stargate.web.models.GetResponseWrapper;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+// JUnit 5 requires both annotations, see
+// https://stackoverflow.com/questions/63250350/junit-5-enabledifsystemproperty-doesnt-work-as-expected
+@EnabledIfSystemProperty(named = "ccm.dse", matches = "true")
+@DisabledIfSystemProperty(named = "ccm.dse", matches = "(?!true)")
+@ExtendWith(CqlSessionExtension.class)
+@CqlSessionSpec(
+    initQueries = {
+      // List
+      "CREATE TABLE lists(k int PRIMARY KEY, l list<int>)",
+      "CREATE CUSTOM INDEX lists_l_idx ON lists(l) USING 'StorageAttachedIndex'",
+      "INSERT INTO lists (k,l) VALUES (1, [1,2,3])",
+      // Set
+      "CREATE TABLE sets(k int PRIMARY KEY, s set<int>)",
+      "CREATE CUSTOM INDEX sets_s_idx ON sets(s) USING 'StorageAttachedIndex'",
+      "INSERT INTO sets (k,s) VALUES (1, {1,2,3})",
+      // Map, indexed by key
+      "CREATE TABLE maps_per_key(k int PRIMARY KEY, m map<int, text>)",
+      "CREATE CUSTOM INDEX maps_per_key_m_idx ON maps_per_key(keys(m)) USING 'StorageAttachedIndex'",
+      "INSERT INTO maps_per_key (k,m) values (1, {1:'a',2:'b',3:'c'})",
+      // Map, indexed by value
+      "CREATE TABLE maps_per_value(k int PRIMARY KEY, m map<int, text>)",
+      "CREATE CUSTOM INDEX maps_per_value_m_idx ON maps_per_value(m) USING 'StorageAttachedIndex'",
+      "INSERT INTO maps_per_value (k,m) values (1, {1:'a',2:'b',3:'c'})",
+      // Map, indexed by entry
+      "CREATE TABLE maps_per_entry(k int PRIMARY KEY, m map<int, text>)",
+      "CREATE CUSTOM INDEX maps_per_entry_m_idx ON maps_per_entry(entries(m)) USING 'StorageAttachedIndex'",
+      "INSERT INTO maps_per_entry (k,m) values (1, {1:'a',2:'b',3:'c'})",
+    })
+public class RestApiv2DseTest extends BaseOsgiIntegrationTest {
+
+  private static final ObjectMapper objectMapper =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+  private static String authToken;
+  private String keyspaceUri;
+
+  @BeforeAll
+  public static void beforeAll(StargateConnectionInfo cluster) throws IOException {
+    authToken = fetchAuthToken(cluster.seedAddress());
+  }
+
+  @BeforeEach
+  public void beforeEach(StargateConnectionInfo cluster, @TestKeyspace CqlIdentifier keyspaceId) {
```

Comment:
After #313 there will always be a separate keyspace per method, and it will be illegal to inject `CqlIdentifier` in `beforeAll`.

Reply:
I've reverted that change on the other PR.